File: 1068.py
while True:
    try:
        entrada = input()
        balance = 0
        correct = True

        for c in entrada:
            if c == '(':
                balance += 1
            elif c == ')':
                balance -= 1
                if balance < 0:
                    correct = False
                    break

        if balance != 0:
            correct = False

        print("correct" if correct else "incorrect")

    except EOFError:
        break

File: 1068.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    string entrada;
    int balance, correct, i;

    while(cin >> entrada) {
        balance = 0;
        correct = 1;

        for(i = 0; entrada[i] != '\0'; i++) {
            if(entrada[i] == '(')
                balance++;
            else if(entrada[i] == ')') {
                balance--;
                if(balance < 0) {
                    correct = 0;
                    break;
                }
            }
        }

        if(balance != 0)
            correct = 0;

        if(correct == 1)
            cout << "correct\n";
        else
            cout << "incorrect\n";
    }


    return 0;
}
